projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.spec.ts

```diff
@@ -91,5 +119,91 @@ describe('OnNavigateFocusService', () => {
       expect(spyDoc).not.toHaveBeenCalledWith('cx-storefront');
       expect(spyEl).not.toHaveBeenCalledTimes(1);
     });
+
+    it('should focus cx-storefront element on navigation', async () => {
+      service.setResetFocusOnNavigate(true);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).toHaveBeenCalledWith('cx-storefront');
+      expect(spyEl).toHaveBeenCalledTimes(1);
+    });
+
+    it('should focus cx-storefront element given breakpoint value', async () => {
+      const breakpointService = TestBed.inject(BreakpointService);
+      spyOn(breakpointService, 'isDown').and.returnValue(of(true));
+
+      service.setResetFocusOnNavigate(BREAKPOINT.sm);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).toHaveBeenCalledWith('cx-storefront');
+      expect(spyEl).toHaveBeenCalledTimes(1);
+    });
+
+    it('should NOT focus cx-storefront element given breakpoint value is not met', async () => {
+      const breakpointService = TestBed.inject(BreakpointService);
+      spyOn(breakpointService, 'isDown').and.returnValue(of(false));
+
+      service.setResetFocusOnNavigate(BREAKPOINT.sm);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).not.toHaveBeenCalledWith('cx-storefront');
+      expect(spyEl).not.toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('setResetViewOnNavigate()', () => {
+    let element: HTMLElement;
+    let spyDoc: jasmine.Spy;
+    let spyEl: jasmine.Spy;
+
+    beforeEach(() => {
+      service = TestBed.inject(OnNavigateFocusService);
+      router = TestBed.inject(Router);
+      zone = TestBed.inject(NgZone);
+
+      element = document.createElement('cx-storefront');
+      spyDoc = spyOn(document, 'getElementsByTagName').and.returnValue([
+        element,
+      ] as any);
+      spyEl = spyOn(element, 'scrollIntoView');
+    });
+
+    it('should scrollIntoView cx-storefront element on navigation', async () => {
+      service.setResetViewOnNavigate(true);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).toHaveBeenCalledWith('cx-storefront');
+      expect(spyEl).toHaveBeenCalledTimes(1);
+    });
+
+    it('should NOT scrollIntoView cx-storefront element on navigation when disabled', async () => {
+      service.setResetViewOnNavigate(false);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).not.toHaveBeenCalledWith('cx-storefront');
+      expect(spyEl).not.toHaveBeenCalledTimes(1);
+    });
+
+    it('should scrollIntoView cx-storefront element given breakpoint value', async () => {
+      const breakpointService = TestBed.inject(BreakpointService);
+      spyOn(breakpointService, 'isDown').and.returnValue(of(true));
+
+      service.setResetViewOnNavigate(BREAKPOINT.sm);
+      await zone.run(() => router.navigateByUrl('/'));
+
+      expect(spyDoc).toHaveBeenCalledWith('cx-storefront');
```

Comment:
We are re-using this string (cx-storefront) in few places in this file, maybe can we move it to some const variable?